### DISCUSSION TITLE
Creating events from custom dataset

### DISCUSSION THREAD

**Discussion by AimanKhan1997** [zach401/acnportal#105] [Q&A]
Hi @sunash and @zach401 . Hope you guys are well and thanks for your help so far. I am working on creating events with my custom dataset and am having issues with creating events that can be passed in ACN-Sim. Can you please let me know if there is an easy way of using generate events using csv file. Also do you prefer using generate_events from acn_data or from stochastic_network?
The dataset looks something like this.
![image](https://user-images.githubusercontent.com/72418210/172051331-43c6120f-9310-4b64-9e70-0eb3d44ce5c6.png)

**Comment by sunash** [comment]
Hi @AimanKhan1997, from the data file you shared, it looks like you'll have to generate the events manually. generat_events in acndata_events is used to generate events from ACN-Data, i.e., already recorded charging events. generate_events in the stochastic_network generates events from a pre-trained gaussian mixture model. For a custom data set, you'll want to load the csv into a pandas DataFrame or similar, and then extract the relevant parameters to create Event instances for each entry in the DataFrame. Let me know if this makes sense to you or if you have any trouble doing this.

**Reply by AimanKhan1997** [reply]
Hi @sunash , thank you for your reply. I managed to create manual events in the following way. Please take a look and let me know your suggestions. The issue I am facing is with departure times, as I have to hardcode it be 30 minutes. Is there any way to get departure times directly from the simulator based on when the vehicle leaves? 

Moreover, regarding the simulator, I believe it takes in every event and runs until every charging session is fulfilled. Is there any way to get a snapshot of the simulator, for example after 45 mins we want to see how many vehicles are charging, how many left and how many are in waiting queue. Please let me know if there is a way to achieve this and how.

Thanks again for your time and consideration. 
````
def event_generator(vr_id, arrival_time, requested_energy,
                  max_rate=200, voltage=750):
    
    battery = acnsim.models.Battery(df_events.iloc[i,3], 0, 200)
    departure_time = arrival_time + 1800
    ev = acnsim.models.EV(arrival_time, departure_time, requested_energy,
                    vr_id, vr_id , battery)
    out_event = acnsim.events.PluginEvent(arrival_time, ev)
    return out_event

event_list = []
for i in range(len(df_events)):
    event_list.append(event_generator(df_events.iloc[i,0], df_events.iloc[i,2], df_events.iloc[i,7]))

new_events = acnsim.events.EventQueue(event_list)
`````
